packages/flutter/test/widgets/scroll_physics_test.dart

```diff
@@ -30,11 +30,11 @@ class TestScrollPhysics extends ScrollPhysics {
 
 void main() {
   test('ScrollPhysics applyTo()', () {
-    const ScrollPhysics a = TestScrollPhysics(name: 'a');
-    const ScrollPhysics b = TestScrollPhysics(name: 'b');
-    const ScrollPhysics c = TestScrollPhysics(name: 'c');
-    const ScrollPhysics d = TestScrollPhysics(name: 'd');
-    const ScrollPhysics e = TestScrollPhysics(name: 'e');
+    const TestScrollPhysics a = TestScrollPhysics(name: 'a');
+    const TestScrollPhysics b = TestScrollPhysics(name: 'b');
+    const TestScrollPhysics c = TestScrollPhysics(name: 'c');
+    const TestScrollPhysics d = TestScrollPhysics(name: 'd');
+    const TestScrollPhysics e = TestScrollPhysics(name: 'e');
```

Comment:
I'm surprised this isn't allowed - isn't `ScrollPhysics` the base class of `TestScrollPhysics`?

Reply:
It is. I made this change because unlike `TestScrollPhysics.applyTo(o)` that returns a `TestScrollPhysics` `ScrollPhysics.applyTo(o)` returns a `ScrollPhysics`. This `applyTo` is used bellow and its result is assign to a `TestScrollPhysics` variable.